cmd/monaco/download/downloaders.go

```diff
@@ -0,0 +1,44 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package download
+
+import (
+	v2 "github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2"
+	projectv2 "github.com/dynatrace/dynatrace-configuration-as-code/pkg/project/v2"
+)
+
+// Downloader represents a component that is responsible for downloading configuration for a given project from Dynatrace
+type Downloader interface {
+
+	// Download downloads configurations from a Dynatrace environment.
+	// If only projectName is given, it will download all configuration.
+	// If additionally specific configuration names/types are given, then it will only download those
+	Download(projectName string, specificConfigs ...string) (projectv2.ConfigsPerType, error)
+}
+
+// Downloaders is a map of downloaders per type of configuration
+type Downloaders map[v2.TypeId]Downloader
+
+// For returns the downloader for a given type of configuration or
+// panics if there is none
+func (d Downloaders) For(t v2.TypeId) Downloader {
+	if dl, ok := d[t]; !ok {
+		panic("No downloader implementation found for type " + t)
+	} else {
```

Comment:
Personally, I am not a fan of unnecessary 'else' branches.
I would just skip the keyword if they are not necessary. 
Absolut nitpick, no need to fix it here

Reply:
Yeah, I would be a bit worried if there a fans of unnecessary "else" branches out there :-O :smiley: